dataquest-dev/DSpace
# feature/de-5-olac

| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |    10.4 |     0 |      0 |        0 |
| Developing      |  0  |    1 |    0 |      0 |         0 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |

## How to test it
OAI shows metadata of items. After adding items or modifying metadata of old items, command `dspace oai import -c` must be called (in _[dspace-installation]/bin_). 

For installation deployed at _dev-5_ [use this guide](https://github.com/dataquest-dev/DSpace/wiki/Dataquest#reindex-github-action)

OAI web interface can be found on _[dev-5.pc/server/oai/](http://dev-5.pc/server/oai/)_

It should be sufficient to use default context. In it, click list metadata formats.

![image](https://user-images.githubusercontent.com/88670521/182641416-04251789-38b2-4fb3-8c81-9e4c893bc2f0.png)

There should be olac format. Click on *List records*. 
![image](https://user-images.githubusercontent.com/88670521/182644024-ac3a2a68-b824-4d8f-8cef-abb56da4dbe7.png)
Under metadata, there are olac-formatted metadata.
![image](https://user-images.githubusercontent.com/88670521/182644283-7eadcb52-28ca-41d4-a45f-519de1e125b8.png)
They should be equivalent to corresponding place in lindat installation. It should never fail or throw exception. 
Olac format is specified here: https://github.com/dataquest-dev/DSpace/blob/feature/de-5-olac/dspace/config/crosswalks/oai/metadataFormats/olac.xsl
Filters (for given context) are specified here: https://github.com/dataquest-dev/DSpace/blob/feature/de-5-olac/dspace/config/crosswalks/oai/xoai.xml

Good luck with all that.
